sema4ai/vscode-client/src/actionServer.ts

```diff
@@ -120,9 +121,21 @@ const startActionServerInternal = async (directory: Uri) => {
         return;
     }
 
+    const home = await getRobocorpHome();
     const env = createEnvWithRobocorpHome(await getRobocorpHome());
     env["RC_ADD_SHUTDOWN_API"] = "1";
 
+    const externalApiPid = path.join(home, "sema4ai-studio", "external-api-server.pid");
```

Comment:
Please make a separate function in the python backend to read this information and call it from here.

It should also make sure that the pid is actually running and connect to the port to know it's listening.

Also, you want to set the same env variable in the `updateLaunchEnvironmentPart0` call (which will have the value on a launch without the action server).